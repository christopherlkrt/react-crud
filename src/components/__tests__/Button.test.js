import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'
import Button from '../Button'

describe('<Button />', () => {
  it('render and generate snapshot', () => {
    const component = renderer.create(<Button disabled />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('testing click', () => {
    /* global jest */
    const handler = jest.fn()
    const button = shallow(<Button onClick={handler} />)
    button.find('button').simulate('click')
    expect(handler.mock.calls.length).toEqual(0)
  })
})
