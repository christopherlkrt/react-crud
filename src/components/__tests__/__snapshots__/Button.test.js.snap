// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> render and generate snapshot 1`] = `
<button
  disabled={true}
  type="button"
>
   
  isDisabled
</button>
`;
